tests/test_selector.py

```diff
@@ -90,9 +90,12 @@ def test_representation_slice(self):
         body = u"<p><input name='{}' value='\xa9'/></p>".format(50 * 'b')
         sel = self.sscls(text=body)
 
-        representation = "<Selector xpath='//input/@name' data='{}'>".format(40 * 'b')
         if six.PY2:
             representation = "<Selector xpath='//input/@name' data=u'{}'>".format(40 * 'b')
+        elif six.PY34:
+            representation = "<Selector xpath='//input/@name' data='{}'>".format('...')
```

Comment:
+1 for writing our own function to shorten the word. Reason is that
```
>>> textwrap.shorten("hello world", width=10, placeholder="...")
'hello...'
>>> textwrap.shorten("helloworld", width=9, placeholder="...")
'...'
```
So what `textwrap.shorten` does is that, `if len(data) > width` it truncates `data` till the nearest whitespace and appends `...` to it
As you said, it becomes an issue when `data` doesn't have whitespaces.